### ISSUE TITLE
Unhelpful error message when trying to use get-pip with Python 2.7 on hiwifi 4 Pro

### ISSUE THREAD

**Issue by q629988171** [pypa/get-pip#42]
Unhelpful error message when trying to use get-pip with Python 2.7 on hiwifi 4 Pro

CPU：MT7621AT + MT7612EN + 7603EN
Mem：256MB DDR3 RAM
Flash：128MB NAND flash
Nic：1+3 x 1000M Ethernet
USB：1x USB 2.0 port. 1x USB 3.0 port
Mainboard：UART pad on PCB (JP3: TX, RX, GND, 3.3V)
OS：HIWIFI OS（ OpenWrt  Barrier Breaker 14.07）

```
root@Hiwifi:/tmp# python get-pip.py 
Traceback (most recent call last):
  File "get-pip.py", line 21373, in <module>
    main()
  File "get-pip.py", line 197, in main
    bootstrap(tmpdir=tmpdir)
  File "get-pip.py", line 82, in bootstrap
    import pip._internal
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/__init__.py", line 40, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/cli/autocompletion.py", line 8, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/cli/main_parser.py", line 12, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/commands/__init__.py", line 6, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/commands/completion.py", line 6, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/cli/base_command.py", line 19, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/download.py", line 37, in <module>
  File "/tmp/tmpiuvLZO/pip.zip/pip/_internal/utils/glibc.py", line 3, in <module>
  File "/tmp/storage/sda1/soft/usr/lib/python2.7/ctypes/__init__.py", line 10, in <module>
    from _ctypes import Union, Structure, Array
ImportError: dynamic module does not define init function (init_ctypes)


when i try 2.6/get-pip.py
root@Hiwifi:/tmp# curl -k https://bootstrap.pypa.io/2.6/get-pip.py | python
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1719k  100 1719k    0     0  33189      0  0:00:53  0:00:53 --:--:-- 21264
Traceback (most recent call last):
  File "<stdin>", line 22123, in <module>
  File "<stdin>", line 194, in main
  File "<stdin>", line 82, in bootstrap
  File "/tmp/tmpna4XUp/pip.zip/pip/__init__.py", line 45, in <module>
  File "/tmp/tmpna4XUp/pip.zip/pip/vcs/mercurial.py", line 9, in <module>
  File "/tmp/tmpna4XUp/pip.zip/pip/download.py", line 36, in <module>
  File "/tmp/tmpna4XUp/pip.zip/pip/utils/glibc.py", line 4, in <module>
  File "/tmp/storage/sda1/soft/usr/lib/python2.7/ctypes/__init__.py", line 10, in <module>
    from _ctypes import Union, Structure, Array
ImportError: dynamic module does not define init function (init_ctypes)
```

**Comment by fcicq** [comment]
newest openwrt is 18.06.

**Comment by pradyunsg** [comment]
Thanks for filing this and for your patience.

This seems to be a case of a Python, that doesn't have a working ctypes module. If you are able to use CPython with a working CPython, it should work.

FWIW, I think the next version of pip drops this expectation from the interpreter -- https://github.com/pypa/pip/pull/6678. It won't be available on Python 2.6 though, since that version has reached EoL (end-of-life).

I'll close this issue eagerly, since I don't there's anything actionable on this issue. Please let me know if there's anything else to do here or if there's additional information that you'd like to share. :)
